src/yunohost/user.py

```diff
@@ -462,6 +469,276 @@ def user_info(auth, username):
+def user_group_update(operation_logger, auth, groupname, add_user=None, remove_user=None, force=False, sync_perm=True):
+    """
+    Update user informations
+
+    Keyword argument:
+        groupname -- Groupname to update
+        add_user -- User to add in group
+        remove_user -- User to remove in group
+
+    """
+
+    from yunohost.permission import permission_sync_to_user
+
+    attrs_to_fetch = ['member']
+
+    if (groupname == 'all_users' or groupname == 'admins') and not force:
+        raise YunohostError('edit_group_not_allowed', group=groupname)
+
+    # Populate group informations
+    result = auth.search(base='ou=groups,dc=yunohost,dc=org',
+                         filter='cn=' + groupname, attrs=attrs_to_fetch)
+    if not result:
+        raise YunohostError('group_unknown', group=groupname)
+    group = result[0]
+
+    new_group_list = {'member': set(), 'memberUid': set()}
+    if 'member' in group:
+        new_group_list['member'] = set(group['member'])
+    else:
+        group['member'] = []
+
+    user_l = user_list(auth, ['uid'])['users']
+
+    if add_user:
+        if not isinstance(add_user, list):
+            add_user = [add_user]
+        for user in add_user:
+            if not user in user_l:
+                raise YunohostError('user_unknown', user=user)
+            userDN = "uid=" + user + ",ou=users,dc=yunohost,dc=org"
+            if userDN in group['member']:
+                logger.warning(m18n.n('user_already_in_group', user=user, group=groupname))
+            new_group_list['member'].add(userDN)
+
+    if remove_user:
+        if not isinstance(remove_user, list):
+            remove_user = [remove_user]
+        for user in remove_user:
+            userDN = "uid=" + user + ",ou=users,dc=yunohost,dc=org"
+            if user == groupname:
+                raise YunohostError('remove_user_of_group_not_allowed', user=user, group=groupname)
+            if 'member' in group and userDN in group['member']:
+                new_group_list['member'].remove(userDN)
+            else:
+                logger.warning(m18n.n('user_not_in_group', user=user, group=groupname))
+
+    # Sychronise memberUid with member (to keep the posix group structure)
+    # In posixgroup the main group of each user is only written in the gid number of the user
+    for member in new_group_list['member']:
+        member_Uid = member.split("=")[1].split(",")[0]
+        # Don't add main user in the group.
+        # Note that in the Unix system the main user of the group is linked by the gid in the user attribute.
+        # So the main user need to be not in the memberUid list of his own group.
+        if member_Uid != groupname:
+            new_group_list['memberUid'].add(member_Uid)
+
+    operation_logger.start()
+
+    if new_group_list['member'] != set(group['member']):
+        if not auth.update('cn=%s,ou=groups' % groupname, new_group_list):
+            raise YunohostError('group_update_failed')
```

Comment:
Ideally it would be nice to include details on why it failed otherwise we won't be able to debug :/ (Maybe looking at similar action, I dunno if we can easily get details on why it failed)